### PR TITLE
Fix #22466 Add github actions dependency updates with dependabot

### LINKED ISSUE #22466: [Feature Request]: Update github actions automatically with dependabot
### What would you like to happen?

Beam currently uses github actions for its CI tasks. Currently most of these tasks are not being updated and we can do it automatically with dependabot.

### Issue Priority

Priority: 3

### Issue Component

Component: build-system

### PR DESCRIPTION
**Pull request by iemejia**
R: @damccorm @pabloem 

**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Review by damccorm — CHANGES_REQUESTED** [review]
I'm -1 on doing this. Right now, infra has to bless each version of GitHub actions that we run (e.g. https://issues.apache.org/jira/browse/INFRA-23219). So that means any time dependabot does an update, a corresponding infra ticket needs to be opened - that doesn't seem worth the effort; I'd rather just upgrade if a bug or feature forces it.

FWIW that's why I didn't initially include actions here.

**Comment by damccorm** [comment]
The only exception to that is that infra does allow actions from the github/actions org automatically. So if we wanted, we could turn it on just for those actions (which would mostly just be the setup-* actions)

**Comment by iemejia** [comment]
I see, a pity that this requires to contact INFRA for simple maintenance. What about the approved actions from other vendors (not github) Can we add those too or do they mind about versions too?

I created a list to enable updates, let me know what you think. Worse case we let only the github ones that are the majority of what Beam currently uses (70 of 91).

**Comment by iemejia** [comment]
Ok I let only the actions from github. I suppose most of them should be already enable for the org and otherwise we might request them. WDYT? Should we give it a try?

**Review by damccorm — APPROVED** [review]
This LGTM now, thanks!

File: .github/dependabot.yml [hunk 1]
```diff
@@ -42,3 +42,10 @@ updates:
     - dependency-name: "com.google.api.grpc:grpc-*"
     - dependency-name: "com.google.http-client:*"
     - dependency-name: "com.google.apis:google-api-services-*"
+  - package-ecosystem: "github-actions"
+    directory: "/"
+    schedule:
+      interval: "daily"
+    allow:
+      # Allow all official github actions
```
Comment (damccorm) [inline .github/dependabot.yml:50]: Optional - might be helpful to specifically call out the approval limitation for future readers

**Comment by iemejia** [comment]
Thanks for the review. I pushed it manually just with the extra suggested comment.

**Pull request closed without merging**